crates/editor/src/editor.rs

```diff
@@ -10038,45 +10048,67 @@ impl Editor {
     }
 
     fn refresh_active_diagnostics(&mut self, cx: &mut ViewContext<Editor>) {
-        if let Some(active_diagnostics) = self.active_diagnostics.as_mut() {
-            let buffer = self.buffer.read(cx).snapshot(cx);
-            let primary_range_start = active_diagnostics.primary_range.start.to_offset(&buffer);
-            let is_valid = buffer
```

Comment:
Right, so this just got simplified out due to how I added support for showing all diagnostics. The basic idea here is a standard set update. I have an old set of diagnostics (whatever is currently being shown) and the new set (whatever the LSP has just sent us). The idea is only hide diagnostics that no longer exist while only adding newly encountered diagnostics. The standard optimization over "hide all old ones, show all new ones" sort of optimization.

The `is_valid` is just doing the same thing except since it only ever cared about a single diagnostic group at once its doing the "what to remove/what to add" logic directly at the individual diagnostic level rather than at the group_id level that I added for showing all diagnostics. I just reused the same logic since it simplified things a bit at the cost of storing that list in the `self.active_diagnostics` HashMap.